README_mps.md

```diff
@@ -0,0 +1,46 @@
+## How to use Stable Diffusion in Apple Silicon (M1/M2)
+
+🤗 Diffusers is compatible with Apple silicon for Stable Diffusion inference, using the PyTorch `mps` device. These are the steps you need to follow to use your M1 or M2 computer with Stable Diffusion.
+
+### Requirements
+
+- Mac computer with Apple silicon (M1/M2) hardware.
+- macOS 12.3 or later.
+- arm64 version of Python.
+
+### Install PyTorch Nightly 
+
+Install the latest [Preview (Nightly) build of PyTorch](https://pytorch.org/get-started/locally/) on your Apple silicon Mac. Please, make sure it meets the requirements above.
+
+We have tested inference using PyTorch version `1.13.0.dev20220830`. After installing, please verify that your version is at least that one.
+
+### Inference Pipeline
+
+The snippet shown below demonstrates how to use the `mps` backend using the familiar `to()` interface to move the Stable Diffusion pipeline to your M1 or M2 device.
+
+We recommend to "prime" the pipeline using an additional one-time pass through it. This is a temporary workaround for a weird issue we have detected: the first inference pass produces slightly different results than subsequent ones. You only need to do this pass once, and it's ok to use just one inference step and discard the result.
+
+```python
+# make sure you're logged in with `huggingface-cli login`
+from diffusers import StableDiffusionPipeline
+
+pipe = StableDiffusionPipeline.from_pretrained("CompVis/stable-diffusion-v1-4", use_auth_token=True)
+pipe = pipe.to("mps")
+
+prompt = "a photo of an astronaut riding a horse on mars"
+
+# First-time "warmup" pass (see explanation above)
+_ = pipe(prompt, num_inference_steps=1)
```

Comment:
Do we need the "warmup" functions at all? They don't seem to be used here. Could we maybe just delete them in this case? For the tests we could do something like:

```python
if torch_device == "mps":
    pipe(prompt, num_inference_steps=1)
```

as well no? 

Think this would actually save quite some code and be cleaner no?

Reply:
Very nice README !